dev/com.ibm.ws.security.wim.adapter.ldap_fat/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/JNDIOutputTest.java

```diff
@@ -0,0 +1,229 @@
+
+        Entry entry = new Entry(SUB_DN);
+        entry.addAttribute("objectclass", "top");
+        entry.addAttribute("objectclass", "domain");
+        ds.add(entry);
+        /*
+         * Add the partition entries.
+         */
+        entry = new Entry("ou=Test,o=ibm,c=us");
+        entry.addAttribute("objectclass", "organizationalunit");
+        entry.addAttribute("ou", "Test");
+        ds.add(entry);
+
+        entry = new Entry(USER_BASE_DN);
+        entry.addAttribute("objectclass", "organizationalunit");
+        entry.addAttribute("ou", "Test");
+        entry.addAttribute("ou", "TestUsers");
+        ds.add(entry);
+
+        entry = new Entry(USER_DN);
+        entry.addAttribute("objectclass", "inetorgperson");
+        entry.addAttribute("uid", USER);
+        entry.addAttribute("sn", USER);
+        entry.addAttribute("cn", USER);
+        entry.addAttribute("userPassword", PWD);
+        ds.add(entry);
+
+    }
+
+    /**
+     * Convenience method to configure the Liberty server with an {@link LdapRegistry} configuration that
+     * will connect to {@link #ldapServer}.
+     *
+     * @throws Exception If there was an error configuring the server.
+     */
+    private static void updateLibertyServer() throws Exception {
+        ServerConfiguration server = emptyConfiguration.clone();
+
+        LdapRegistry ldap = new LdapRegistry();
+        ldap.setId("ldap1");
+        ldap.setRealm("LDAPRealm");
+        ldap.setHost("localhost");
+        ldap.setPort(String.valueOf(ds.getLdapPort()));
+        ldap.setBaseDN(SUB_DN);
+        ldap.setBindDN(InMemoryLDAPServer.getBindDN());
+        ldap.setBindPassword(InMemoryLDAPServer.getBindPassword());
+        ldap.setLdapType("Custom");
+
+        server.getLdapRegistries().add(ldap);
+        updateConfigDynamically(libertyServer, server);
+
+    }
+
+    /**
+     *
+     * Check that we hit the JNDI BER output
+     */
+    @Test
+    public void testJNDIBEROutput() throws Exception {
+        String trace = "-> localhost";
+
+        /*
+         * Run with JNDI BER trace disabled
+         */
+        String returnUser = servlet.checkPassword(USER_DN, PWD);
+        List<String> errMsgs = libertyServer.findStringsInLogsAndTrace(trace);
+        assertTrue("Should not find: " + trace, errMsgs.isEmpty());
+
+        /*
+         * Enable JNDI BER trace output
+         */
+        ServerConfiguration config = libertyServer.getServerConfiguration();
+        ConfigElementList<LdapRegistry> ldaps = config.getLdapRegistries();
+        LdapRegistry ldap = null;
+        if (!ldaps.isEmpty() && ldaps.size() == 1) {
+            ldap = ldaps.get(0);
+        }
+        ldap.setJndiOutputEnabled(true);
+        updateConfigDynamically(libertyServer, config);
+        returnUser = servlet.checkPassword(USER_DN, PWD);
+        errMsgs = libertyServer.findStringsInLogsAndTrace(trace);
+        assertFalse("Should find: " + trace, errMsgs.isEmpty());
```

Comment:
I thought i put a comment about checking for more output, but I don't see it. Is there anything else we can check for? Maybe the incoming and  outgoing, maybe some data?

Reply:
Need to run with several JREs as well. the trace may be different.